tasks/ThrusterLimitationTask.cpp

```diff
@@ -0,0 +1,104 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "ThrusterLimitationTask.hpp"
+#include <algorithm>
+
+using namespace std;
+using namespace base;
+using namespace gazebo_usv;
+using namespace control_base;
+
+ThrusterLimitationTask::ThrusterLimitationTask(std::string const& name)
+    : ThrusterLimitationTaskBase(name)
+{
+}
+
+ThrusterLimitationTask::~ThrusterLimitationTask()
+{
+}
+
+/// The following lines are template definitions for the various state machine
+// hooks defined by Orocos::RTT. See ThrusterLimitationTask.hpp for more detailed
+// documentation about them.
+
+bool ThrusterLimitationTask::configureHook()
+{
+    if (!ThrusterLimitationTaskBase::configureHook()) {
+        return false;
+    }
+    m_limits = _limits.get();
```

Comment:
shouldn't we validate that the max_limit is bigger than the min_limit?
checkSpeedSaturation could go wrong if the limits are defined this way